python/tank/log.py

```diff
@@ -100,69 +212,96 @@ def get_root_logger(cls):
 
         :param log_name: Name of logger to create. This will form the
-                         filename of the log file.
+                         filename of the log file. If you pass None, you
+                         ensure that any existing base file handlers terminate
+                         their logging.
+        :returns: The name of the previous log_name that is being switched away from.
         """
-        # avoid cyclic references
-        from .util import LocalFileStorageManager
-        from .util import filesystem
 
-        if cls._std_file_handler:
+        previous_log_name = self._std_file_handler_log_name
+        log.debug("Switching file based std logger from %s to %s" % (previous_log_name, log_name))
+
+        if self._std_file_handler:
             # there is already a log handler.
             # terminate previous one
-            sgtk_root_logger.removeHandler(cls._std_file_handler)
-            cls._std_file_handler = None
+            log.debug(
+                "Tearing down existing log handler '%s' (%s)" % (previous_log_name, self._std_file_handler)
+            )
+            self._root_logger.removeHandler(self._std_file_handler)
+            self._std_file_handler = None
 
-        # set up logging root folder
-        log_folder = LocalFileStorageManager.get_global_root(LocalFileStorageManager.LOGGING)
-        filesystem.ensure_folder_exists(log_folder)
+        # store new log name
+        self._std_file_handler_log_name = log_name
 
-        # generate log path
-        log_file = os.path.join(
-            log_folder,
-            "%s.log" % filesystem.create_valid_filename(log_name)
-        )
+        if log_name:
+            # set up a new handler
 
-        cls._std_file_handler = logging.handlers.RotatingFileHandler(
-            log_file,
-            maxBytes=4096*1024,
-            backupCount=5
-        )
+            # avoid cyclic references
+            from .util import filesystem
+
+            # set up logging root folder
+            filesystem.ensure_folder_exists(self.log_folder)
+
+            # generate log path
+            log_file = os.path.join(
+                self.log_folder,
+                "%s.log" % filesystem.create_valid_filename(log_name)
+            )
 
-        # example:
-        # 2016-04-25 08:56:12,413 [44862 DEBUG tank.log] message message
-        formatter = logging.Formatter(
-            "%(asctime)s [%(process)d %(levelname)s %(name)s] %(message)s"
-        )
 
-        cls._std_file_handler.setFormatter(formatter)
-        sgtk_root_logger.addHandler(cls._std_file_handler)
+            # create a rotating log file with a max size of 18 megs -
+            # zendesk's current max attachment limit is 20mb so
+            # this should make all log files attachable to support tickets.
+            self._std_file_handler = logging.handlers.RotatingFileHandler(
+                log_file,
+                maxBytes=1024*1024*18,  # 18 MiB
```

Comment:
I think this is a bit high still.  Just for reference a tk-desktop file that was rotated at 1MB comes out to 4477 lines.  If we do make this smaller we're more likely to fit when asking somebody to just "zip up their log dir" so we don't need to ask for individual log files...

Not critical, but a value of 4 or 5 MB feels like plenty.